### PR TITLE
Enabling adding custom tags for repos beyond pytorch/pytorch

### PR DESCRIPTION
**Pull request by ZainRizvi**
This will enable repos such as pytorch/tau to also have their PRs autolabled.

Current support is pretty rudimentary: All labels corresponding to a file path modified in that PR will be added to the PR

We can evolve this implementation as we get requests from other repos

### CODE REVIEW

File: torchci/lib/bot/autoLabelBot.ts [hunk 1]
```diff
@@ -104,6 +104,32 @@ const notUserFacingPatternExceptions: RegExp[] = [
...
+  ]
+}
+
+function getRepoSpecificLabels(owner: string, repo: string): [RegExp, string][] {
+  var repoKey = owner + "/" + repo;
+  if (!repoSpecificAutoLabels.hasOwnProperty(repoKey)) {
+    return []
+  }
+
+  return repoSpecificAutoLabels[repoKey]
```
Comment (janeyx99) [inline torchci/lib/bot/autoLabelBot.ts:130]: nit: inconsistent missing `;` across this change

File: torchci/test/autoLabelBot.test.ts [hunk 2]
```diff
@@ -317,6 +317,37 @@ describe("auto-label-bot", () => {
...
   });
 
+  test("custom repo labels get add when on a matching repo and file", async () => {
+
+    nock("https://api.github.com")
+      .post("/app/installations/2/access_tokens")
+      .reply(200, { token: "test" });
+
+    const payload = requireDeepCopy("./fixtures/pull_request.opened")["payload"];
+    payload["pull_request"]["title"] = "modify a pytorch/tau file";
```
Comment (janeyx99) [inline torchci/test/autoLabelBot.test.ts:327]: isn't this a fake-test-repo file?
```suggestion
    payload["pull_request"]["title"] = "modify a pytorch/fake-test-repo file";
```

**Comment by janeyx99** [comment]
Was this a particular customer req?

**Comment by ZainRizvi** [comment]
> Was this a particular customer req?

Yeah, the pytorch/tau folks wanted to be able to auto-label their PRs

**Review by janeyx99 — APPROVED** [review]
For now this is fine/functional, but for better organizational/cleanliness purposes, we should eventually have different yamls for different repo patterns or something less confusing than this.

I can see things like:
pytorch.yml
tau.yml
common.yml

It would require a bit more thought, though, since for things like categories/topics we have more advanced ideas than just "if it applies, just add label". We would need to think about label priority and exclusivity. Could be interesting.

**Comment by ZainRizvi** [comment]
> For now this is fine/functional, but for better organizational/cleanliness purposes, we should eventually have different yamls for different repo patterns or something less confusing than this.
> 
> I can see things like: pytorch.yml tau.yml common.yml
> 
> It would require a bit more thought, though, since for things like categories/topics we have more advanced ideas than just "if it applies, just add label". We would need to think about label priority and exclusivity. Could be interesting.

+1 to that long term goal, this is definitely a quick-n-dirty approach. I wanted to get something out the door to meet this customer request for now and then wait to see what shape future requests take before building out something more generalized

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
> For now this is fine/functional, but for better organizational/cleanliness purposes, we should eventually have different yamls for different repo patterns or something less confusing than this.
> 
> I can see things like: pytorch.yml tau.yml common.yml
> 
> It would require a bit more thought, though, since for things like categories/topics we have more advanced ideas than just "if it applies, just add label". We would need to think about label priority and exclusivity. Could be interesting.

To elaborate on this idea - it should not be a property of autoLabel bot, but rather a function of  target repository. As we already have the logic for reading a repo specific yaml, in `autoCCbot.ts`, see
https://github.com/pytorch/test-infra/blob/b7d40f820e2612d620569e2f748f55353c342dff/torchci/lib/bot/utils.ts#L26

**Pull request merged**